engine/src/test/scala/engine/workflow/ValidateActorSpec.scala

```diff
@@ -0,0 +1,177 @@
+                if (validationException.errors.size != 1) fail("Number of errors coming from validation exception are not the expected ones.")
+                if (!validationException.errors.head.contains("Unable to load namespace from workflow")) fail("Message from error nr 1 in validation exception is not the expected one.")
+            }
+          case unknown => fail(s"Response is not equals to the expected one. Response: $unknown")
+        }
+      }
+      verify(backendMock, times(1)).assertWorkflowOptions(WorkflowOptions.fromJsonString(validOptionsJson).get)
+    }
+
+    "return ValidationFailure when there are workflow options coming from a malformed workflow options JSON file" in {
+      within(1000 millis) {
+        validateActor ! ValidateWorkflow(validWdlSource, Some(validInputsJson), Some(malformedOptionsJson))
+        expectMsgPF() {
+          case validationFailure: ValidationFailure =>
+            validationFailure.reason match {
+              case validationException: ValidationException =>
+                if (!validationException.message.contains("Workflow input processing failed.")) fail("Message coming from validation exception is not the expected one.")
+                if (validationException.errors.size != 1) fail("Number of errors coming from validation exception are not the expected ones.")
+                if (!validationException.errors.head.contains("Workflow contains invalid options JSON")) fail("Message from error nr 1 in validation exception is not the expected one.")
+            }
+          case unknown => fail(s"Response is not equals to the expected one. Response: $unknown")
+        }
+      }
+      verify(backendMock, times(0)).assertWorkflowOptions(WorkflowOptions.fromJsonString(validOptionsJson).get)
+    }
+
+    "return ValidationFailure when there are invalid workflow options coming from a workflow options JSON file" in {
+      when(backendMock.assertWorkflowOptions(WorkflowOptions.fromJsonString(validOptionsJson).get)).thenThrow(new IllegalStateException("Some exception"))
+      within(1000 millis) {
+        validateActor ! ValidateWorkflow(validWdlSource, Some(validInputsJson), Some(validOptionsJson))
+        expectMsgPF() {
+          case validationFailure: ValidationFailure =>
+            validationFailure.reason match {
+              case validationException: ValidationException =>
+                if (!validationException.message.contains("Workflow input processing failed.")) fail("Message coming from validation exception is not the expected one.")
+                if (validationException.errors.size != 1) fail("Number of errors coming from validation exception are not the expected ones.")
+                if (!validationException.errors.head.contains("Workflow has invalid options for backend")) fail("Message from error nr 1 in validation exception is not the expected one.")
```

Comment:
It just validates part of that message. I'm just validating the part of the validation functions based on parts of the messages that identifies that the validation went through that function.